File: client-clojure/src/sirqul_io_t_platform/specs/postal_code.clj
(ns sirqul-io-t-platform.specs.postal-code
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.region :refer :all]
            )
  (:import (java.io File)))


(def postal-code-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :locationDescription) string?
   (ds/opt :code) string?
   (ds/opt :plus4) string?
   (ds/opt :stateCode) string?
   (ds/opt :city) string?
   (ds/opt :timeZone) string?
   (ds/opt :utcOffset) int?
   (ds/opt :regions) (s/coll-of region-spec)
   (ds/opt :eliminated) boolean?
   (ds/opt :searchPoint) boolean?
   (ds/opt :lastOfferImport) inst?
   (ds/opt :truncatedLatitude) float?
   (ds/opt :truncatedLongitude) float?
   })

(def postal-code-spec
  (ds/spec
    {:name ::postal-code
     :spec postal-code-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/PostalCode.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import PostalCode._

case class PostalCode (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
latitude: Option[Double],
longitude: Option[Double],
locationDescription: Option[String],
code: Option[String],
plus4: Option[String],
stateCode: Option[String],
city: Option[String],
timeZone: Option[String],
utcOffset: Option[Integer],
regions: Option[Set[Region]],
eliminated: Option[Boolean],
searchPoint: Option[Boolean],
lastOfferImport: Option[OffsetDateTime],
truncatedLatitude: Option[Double],
truncatedLongitude: Option[Double])

object PostalCode {
  import DateTimeCodecs._

  implicit val PostalCodeCodecJson: CodecJson[PostalCode] = CodecJson.derive[PostalCode]
  implicit val PostalCodeDecoder: EntityDecoder[PostalCode] = jsonOf[PostalCode]
  implicit val PostalCodeEncoder: EntityEncoder[PostalCode] = jsonEncoderOf[PostalCode]
}
